channels/urbdrc/client/data_transfer.c

```diff
@@ -21,419 +21,361 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <unistd.h>
 
-#include <sys/ioctl.h>
-#include <sys/stat.h>
-#include <sys/types.h>
+#include <winpr/sysinfo.h>
+
+#include <urbdrc_helpers.h>
 
 #include "urbdrc_types.h"
 #include "data_transfer.h"
 
-static void usb_process_get_port_status(IUDEVICE* pdev, BYTE* OutputBuffer)
+static void usb_process_get_port_status(IUDEVICE* pdev, wStream* out)
 {
 	int bcdUSB = pdev->query_device_descriptor(pdev, BCD_USB);
 
 	switch (bcdUSB)
 	{
 		case USB_v1_0:
-			data_write_UINT32(OutputBuffer, 0x303);
+			Stream_Write_UINT32(out, 0x303);
 			break;
 
 		case USB_v1_1:
-			data_write_UINT32(OutputBuffer, 0x103);
+			Stream_Write_UINT32(out, 0x103);
 			break;
 
 		case USB_v2_0:
-			data_write_UINT32(OutputBuffer, 0x503);
+			Stream_Write_UINT32(out, 0x503);
 			break;
 
 		default:
-			data_write_UINT32(OutputBuffer, 0x503);
+			Stream_Write_UINT32(out, 0x503);
 			break;
 	}
 }
 
-#if ISOCH_FIFO
-
-static int func_check_isochronous_fds(IUDEVICE* pdev)
+static UINT urb_write_completion(IUDEVICE* pdev, URBDRC_CHANNEL_CALLBACK* callback, BOOL noAck,
+                                 wStream* out, UINT32 InterfaceId, UINT32 MessageId,
+                                 UINT32 RequestId, UINT32 usbd_status, UINT32 OutputBufferSize)
 {
-	int ret = 0;
```

Comment:
I'd suggest `out` to be `wStream **` instead of `wStream *`, so that this function can clear the pointer reference.
Perhaps this stream could also come from a `StreamPool`, as we will call it multiple times with stream with quite the same size...